apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.cs

```diff
@@ -239,6 +243,59 @@ internal UrlSigner WithSigningVersion(SigningVersion signingVersion)
                 request?.Headers?.ToDictionary(h => h.Key, h => h.Value),
                 request?.Content?.Headers?.ToDictionary(h => h.Key, h => h.Value));
 
+        /// <summary>
+        /// Creates a signed URL which can be used to provide limited access to specific buckets and objects to anyone
+        /// in possession of the URL, regardless of whether they have a Google account.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// When a <see cref="UrlSigner"/> is created with a service account credential, the signing can be performed
+        /// with no network access. When it is created with an implementation of <see cref="IBlobSigner"/>, that may require
+        /// network access or other IO. In that case, one of the asynchronous methods should be used when the caller is
+        /// in a context that should not block.
+        /// </para>
+        /// <para>
+        /// When the <paramref name="request"/> is specified, some of its headers will be included in the signed URL's
+        /// signature, and therefore must be included in requests made with the created URL. These are the Content-MD5 and
+        /// Content-Type content headers as well as any content or request header with a name starting with "x-goog-".
+        /// </para>
+        /// <para>
+        /// If <paramref name="request"/> is null, no headers are included in the signed URL's signature, so any requests
+        /// made with the created URL must not contain Content-MD5, Content-Type, or any header starting with "x-goog-".
+        /// </para>
+        /// <para>
+        /// Note that if the entity is encrypted with customer-supplied encryption keys (see
+        /// https://cloud.google.com/storage/docs/encryption for more information), the <b>x-goog-encryption-algorithm</b>,
+        /// <b>x-goog-encryption-key</b>, and <b>x-goog-encryption-key-sha256</b> headers will be required when making the
+        /// request. However, only the x-goog-encryption-algorithm header is included in the signature for the signed URL.
```

Comment:
I might be wrong, but a developer could include `x-goog-encryption-key` into Signed Headers or is this more about security concerns with CSEK?

Reply:
We explicitly don't include x-goog-encryption-key in the signed headers, as per step 3 of https://cloud.google.com/storage/docs/access-control/signed-urls#about-canonical-extension-headers

Reply:
Nice! I missed that. Thanks @jskeet!